compiler/rustc_borrowck/src/type_check/constraint_conversion.rs

```diff
@@ -83,16 +83,8 @@ impl<'a, 'tcx> ConstraintConversion<'a, 'tcx> {
         }
         self.constraints.member_constraints = tmp;
 
-        for (predicate, constraint_category) in outlives {
-            // At the moment, we never generate any "higher-ranked"
-            // region constraints like `for<'a> 'a: 'b`. At some point
-            // when we move to universes, we will, and this assertion
-            // will start to fail.
```

Comment:
this assumption is probably why we've had a binder here in the first place.

I believe this to be wrong. We've already mostly moved to universes. I think we want to avoid mapping placeholders back to bound variables inside of the solver so even in the future we won't deal with bound vars here. And finally I expect us to move higher ranked region solving into the trait solver at some point in the future to deal with implications in binders. At that point we not only not return higher ranked outlives predicates from queries, we completely stop returning placeholders from a binder in the query.